### PR TITLE
Add Forms to Blog App 📝✨

### PR DESCRIPTION
**Pull request by DevKaranJ**
👋 I've successfully added forms to our Blog app, completing the designated tasks. Here's a brief overview of the changes:

- Implemented a current_user method in the ApplicationController to provide access to the first user in the database.
- Created forms for the following functionalities:

- [x] Post creation on behalf of the current_user.
- [x] Comment creation on behalf of the current_user.
- [x] Allowing users to add likes to posts.

I ensured that the code adheres to Ruby best practices and resolved any linter errors. The Gitflow process has been followed diligently.

Ready for your review! 🚀🔍

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb]: - [x] @DevKaranJ Great work adding a method to add a new post , however your method doesn't use the current_user method as per the project requirement and the post params is missing the title parameter, kindly consider fixing this error
image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user
+
+    return unless @comment.save
+
+    redirect_to user_post_path(@post.author, @post), notice: 'Comment created successfully'
+  end
```
Comment (danmainah) [inline app/controllers/comments_controller.rb]: - [x] @DevKaranJ Great work adding a method to add new comments,  however you have not used the `current_user` method as required in the project requirements.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @DevKaranJ , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added the current_user method in the ApplicationController to make current user data available to all controllers.
✔️ Added action methods to create posts, comments, and likes
✔️ Added forms to add new posts, comments and a button to add likes to a post.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
```
Comment (OyePriscilla) [inline app/controllers/posts_controller.rb]: - [x] I was unable to create a new post successfully for the `current_user`, kindly work around the `current_user` implementation properly as suggested earlier. 🙏

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params)
+
+    # Assign the comment to the current_user
+    @comment.user = current_user
+
```
Comment (OyePriscilla) [inline app/controllers/comments_controller.rb:8]: - [x] In the process of creating a comment for a given post for the `current_user`, kindly ensure that comment is successfully created for the current_user as suggested earlier. 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @DevKaranJ 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Included title as post params ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @DevKaranJ 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Posts and comments are successfully being created👍🏼.
- [x] Nice UI👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**